components/support/nimbus-fml/src/workflows.rs

```diff
@@ -149,4 +177,44 @@ mod test {
         generate_and_assert("test/app_menu.kts", "fixtures/ir/app_menu.json", true)?;
         Ok(())
     }
+
+    fn validate_against_experimenter_schema<P: AsRef<Path>>(
+        schema_path: P,
+        generated_json: &serde_json::Value,
+    ) -> Result<()> {
+        let schema = fs::read_to_string(&schema_path)?;
+        let schema: serde_json::Value = serde_json::from_str(&schema)?;
+        let compiled = JSONSchema::compile(&schema).expect("The schema is invalid");
+        let res = compiled.validate(generated_json);
+        if let Err(e) = res {
+            let mut errs: String = "Validation errors: \n".into();
+            for err in e {
+                errs.push_str(&format!("{}\n", err));
+            }
+            panic!("{}", errs);
+        }
+        Ok(())
+    }
+
+    #[test]
+    fn test_schema_validation() -> Result<()> {
```

Comment:
This test takes a noticeably long time to run because of all the file system I/O, I think I'll make the tests avoid writing the generated json to disk (instead validate on an in-memory json), and read the schema exactly once